cli/lib/appdata.ts

```diff
@@ -122,14 +131,24 @@ export async function getAuthToken(): Promise< NonNullable< UserData[ 'authToken
 
 export async function getSiteByFolder(
 	siteFolder: string
-): Promise< z.infer< typeof siteSchema > > {
+): Promise< z.infer< typeof siteSchema > | null > {
 	const userData = await readAppdata();
 	const sites = userData.sites ?? [];
 	const site = sites.find( ( site ) => site.path === siteFolder );
 
 	if ( ! site ) {
-		throw new LoggerError( __( 'The specified folder is not added to Studio.' ) );
+		return null;
```

Comment:
I guess this is ultimately subjective, but I would rather continue throwing an error from this function and instead wrap the call in `saveSnapshotToAppdata` in a `try .. catch`